reconcile/utils/jira_client.py

```diff
@@ -41,9 +41,7 @@ def get_issues(self, fields: Optional[Mapping] = None) -> list[Issue]:
             kwargs["fields"] = ",".join(fields)
         while True:
             index = block_num * block_size
-            issues: Union[dict[str, Any], ResultList[Issue]] = self.jira.search_issues(
-                jql, index, block_size, **kwargs
-            )
+            issues = self.jira.search_issues(jql, index, block_size, **kwargs)
```

Comment:
That's pretty confusing error message, since the type is just copied over from search_issue. 
Wonder if there is some special case, where only a dict is returned? :confused:

Reply:
If I read the error correctly, there are times where just a dict is returned, couldn't that break `all_issues.extend(issues)` ?

Reply:
Okay, got it now. Think we should be safe then. 
What about bumping the dependency?